### PR TITLE
Messages implemented

### PR DESCRIPTION
**Pull request by saraquejido**

### CODE REVIEW

File: src/main/java/cat/udl/eps/softarch/linkapp/LinkAppApplication.java [hunk 1]
```diff
@@ -1,13 +1,40 @@
...
+    public static void main(String[] args) {
+        SpringApplication.run(LinkAppApplication.class, args);
+    }
+
+    @Autowired
+    private MeetRepository meetRepository;
+
+    @Scheduled(cron = "${meet-status-cron}")
+    @Transactional
+    protected void updateMeetStatusCronJob() {
```
Comment (bergacat1) [inline src/main/java/cat/udl/eps/softarch/linkapp/LinkAppApplication.java:28]: Implementing this here is quite strange. You should move it to a @Configuration or @Service class.

File: src/test/resources/features/DeleteMessage.feature [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    Given There is a registered user with username "user" and password "password" and email "user@sample.app"
+    And A group exists with a meet
+
+  Scenario: Delete a Message in a meet I am assisting
+    Given I login as "user" with password "password"
+    And The user "user" belongs to that group as "AUTHORIZED"
+    And The user "user" is assisting to the meet
+    And I send a message to the meet with message "message"
+    When I delete the message
+    Then The response code is 403
```
Comment (bergacat1) [inline src/test/resources/features/DeleteMessage.feature:15]: So it's not possible to remove sent messages?

File: src/main/java/cat/udl/eps/softarch/linkapp/repository/MeetRepository.java [hunk 3]
```diff
@@ -18,4 +19,7 @@
    */
 
   List<Meet> findByTitleContaining(@Param("title") String title);
+  List<Meet> findByStatus(@Param("status") Boolean status);
+  List<Meet> findByStatusAndFinalMeetDateLessThan(@Param("status") Boolean status, @Param("finalMeetDate") ZonedDateTime meetDate);
```
Comment (bergacat1) [inline src/main/java/cat/udl/eps/softarch/linkapp/repository/MeetRepository.java:23]: These methods are not tested

File: src/main/java/cat/udl/eps/softarch/linkapp/repository/MessageRepository.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import org.springframework.data.repository.PagingAndSortingRepository;
+import org.springframework.data.repository.query.Param;
+import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+
+import java.util.List;
+
+@RepositoryRestResource
+public interface MessageRepository extends PagingAndSortingRepository<Message, Long> {
+  public List<Message> findByAuthor(@Param("author") User author);
+  public Message findByIdContaining(@Param("number") Long number);
```
Comment (bergacat1) [inline src/main/java/cat/udl/eps/softarch/linkapp/repository/MessageRepository.java:14]: These methods are not tested.

**Review by bergacat1 — APPROVED** [review]
I accept the PR because generally it's correct. However, I've written some comments that you shoud fix in a new PR.

File: src/test/resources/features/DeleteMessage.feature (hunk 2 above)
Comment (saraquejido) [reply src/test/resources/features/DeleteMessage.feature:15]: It's not possible to remove nor to edit sent messages.

File: src/main/java/cat/udl/eps/softarch/linkapp/LinkAppApplication.java (hunk 1 above)
Comment (alexflorensa) [reply src/main/java/cat/udl/eps/softarch/linkapp/LinkAppApplication.java:28]: In which package should I put this class?

Comment (rogargon) [reply src/main/java/cat/udl/eps/softarch/linkapp/LinkAppApplication.java:28]: Domain logic is usually kept in Controllers and Services. We are using controllers automatically generated by Spring, through RestRepositories, so our choice is Services. The are implemented in a separate class, annotated with `@Service`, like: https://github.com/UdL-EPS-SoftArch/MyTournamentX-API/blob/master/src/main/java/cat/udl/eps/softarch/mytournamentx/service/MatchWinnerService.java

In your case, the service will include the method being schedules via cron

File: src/main/java/cat/udl/eps/softarch/linkapp/repository/MeetRepository.java (hunk 3 above)
Comment (alexflorensa) [reply src/main/java/cat/udl/eps/softarch/linkapp/repository/MeetRepository.java:23]: How should we test these methods? There is no additional logic beyond that defined in the Spring methods.

File: src/main/java/cat/udl/eps/softarch/linkapp/repository/MessageRepository.java (hunk 4 above)
Comment (bcaspi) [reply src/main/java/cat/udl/eps/softarch/linkapp/repository/MessageRepository.java:14]: findByAuthor has been tested when checking that the author of a new message is the correct one. Regarding the other method, we've removed it as it was useless considering that it was never used and we have the findById predefined method.

File: src/main/java/cat/udl/eps/softarch/linkapp/repository/MeetRepository.java (hunk 3 above)
Comment (rogargon) [reply src/main/java/cat/udl/eps/softarch/linkapp/repository/MeetRepository.java:23]: Just use them from the stepdefs, through the API (e.g. `\meets\search\findByStatus`)  or from the MeetRepository instance

**Pull request merged**